internal/benchmarks/benchmarks_test.go

```diff
@@ -0,0 +1,274 @@
+	*TestOperator
+	attacher         Attacher
+	mountNsMapSetter MountNsMapSetter
+	containers       []*containercollection.Container
+	testingTB        testing.TB
+	tc               *tracercollection.TracerCollection
+}
+
+func (t *TestOperatorInstance) PreGadgetRun() error {
+	if t.attacher != nil {
+		for i := range t.containers {
+			err := t.attacher.AttachContainer(t.containers[i])
+			if err != nil {
+				t.testingTB.Fatal(err)
+			}
+		}
+	}
+	if t.mountNsMapSetter != nil {
+		tc, err := tracercollection.NewTracerCollectionTest(nil)
+		if err != nil {
+			t.testingTB.Fatal(err)
+		}
+		t.tc = tc
+		tc.AddTracer("test-tracer", containercollection.ContainerSelector{})
+
+		mountnsmap, err := tc.TracerMountNsMap("test-tracer")
+		if err != nil {
+			t.testingTB.Fatal(err)
+		}
+		t.mountNsMapSetter.SetMountNsMap(mountnsmap)
+	}
+	return nil
+}
+
+func (t *TestOperatorInstance) PostGadgetRun() error {
+	if t.attacher != nil {
+		for i := range t.containers {
+			err := t.attacher.DetachContainer(t.containers[i])
+			if err != nil {
+				t.testingTB.Fatal(err)
+			}
+
+		}
+	}
+	if t.mountNsMapSetter != nil {
+		t.tc.Close()
+	}
+	return nil
+}
+
+func (t *TestOperatorInstance) EnrichEvent(ev any) error {
+	return nil
+}
+
+func init() {
+	operators.Register(&TestOperatorInstance{})
+}
+
+func BenchmarkAllGadgetsManyContainers(b *testing.B) {
+	utilstest.RequireRoot(b)
+
+	const containerCount = 100
+
+	// Prepare runtime
+	runtime := &local.Runtime{}
+	err := runtime.Init(nil)
+	if err != nil {
+		b.Fatalf("initializing runtime: %s", err)
+	}
+	defer runtime.Close()
+
+	// Prepare fake containers
+	runnerConfig := &utilstest.RunnerConfig{}
+	var containers []*containercollection.Container
+	for i := 0; i < containerCount; i++ {
+		runner := utilstest.NewRunnerWithTest(b, runnerConfig)
+		container := &containercollection.Container{
+			ID:    fmt.Sprintf("container%d", i),
+			Mntns: runner.Info.MountNsID,
+			Netns: runner.Info.NetworkNsID,
+			Pid:   uint32(runner.Info.Tid),
+		}
+		containers = append(containers, container)
+	}
+
+	allGadgets := gadgetregistry.GetAll()
+
+	for _, gadgetDesc := range allGadgets {
+		gadgetDesc := gadgetDesc
+
+		// Skip unwanted gadgets
+		categoryAndName := fmt.Sprintf("%s-%s", gadgetDesc.Category(), gadgetDesc.Name())
+		skipRegex := os.Getenv("IG_BENCHMARKS_GADGET_REGEX")
+		if skipRegex != "" {
+			matched, _ := regexp.Match(skipRegex, []byte(categoryAndName))
+			if !matched {
+				continue
+			}
+		}
+
+		if _, ok := gadgetDesc.(gadgets.GadgetInstantiate); !ok {
+			continue
+		}
+
+		validOperators := operators.GetOperatorsForGadget(gadgetDesc)
+		operatorsParamCollection := validOperators.ParamCollection()
+
+		err = validOperators.Init(nil)
+		if err != nil {
+			b.Fatalf("initializing operators: %s", err)
+		}
+		defer validOperators.Close()
+
+		b.Run(categoryAndName, func(b *testing.B) {
+			for n := 0; n < b.N; n++ {
+				ctx, cancel := context.WithTimeout(context.TODO(), 0)
```

Comment:
Do you plan to pass a timeout there?

Reply:
Timeout of 0s is necessary so it immediately stops the gadget once it has been started. I'll add a comment to explain.